### PR TITLE
Fix swin backbone absolute pos_embed

### PR DESCRIPTION
**Pull request by normster**
## Motivation

The Swin transformer backbone is currently unable to process input images differing in size from the pretraining resolution when absolute positional embedding is enabled, due to lack of dynamic pos embed resizing in the forward pass.

## Modification

When absolute positional embedding is enabled, the backbone now checks the input shape against the pos embed shape and performs a bicubic interpolation to the correct shape if necessary. This is based on the original Mask2former code: https://github.com/facebookresearch/Mask2Former/blob/main/mask2former/modeling/backbone/swin.py#L656.

The pos embed parameter's shape has also been changed from [B * (H*W) * D] -> [B * D * H * W] to bring the model into consistency with the checkpoint loading code, which expects the new shape.

## BC-breaking (Optional)

This fix should not break backwards compatibility. No existing configs using the Swin backbone have absolute positional embedding enabled, or load from 

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by chhluo — APPROVED** [review]
LGTM

**Comment by chhluo** [comment]
please add a unit test for the modification in https://github.com/open-mmlab/mmdetection/blob/master/tests/test_models/test_backbones/test_swin.py

File: mmdet/models/backbones/swin.py [hunk 1]
```diff
@@ -746,7 +745,17 @@ def forward(self, x):
...
-            x = x + self.absolute_pos_embed
+            h, w = self.absolute_pos_embed.shape[1:3]
+            if hw_shape[0] != h or hw_shape[1] != w:
+                absolute_pos_embed = F.interpolate(
+                    self.absolute_pos_embed,
+                    size=hw_shape,
+                    mode='bicubic',
+                    align_corners=False).flatten(2).transpose(1, 2)
+            else:
+                absolute_pos_embed = self.absolute_pos_embed\
```
Comment (ZwwWayne) [inline mmdet/models/backbones/swin.py:756]: '\' is unnecessary here.

**Review by ZwwWayne — APPROVED** [review]

**Comment by ZwwWayne** [comment]
This PR needs to be migrated to dev-3.x

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @normster ！First of all, we want to express our gratitude for your significant PR in the mmdetection project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**